clinica/pipelines/t1_freesurfer_longitudinal/t1_freesurfer_template_pipeline.py

```diff
@@ -1,34 +1,38 @@
 # coding: utf8
 
-__author__ = "Alexis Guyot"
-__copyright__ = "Copyright 2016-2019, The Aramis Lab Team"
-__credits__ = ["Alexis Guyot"]
-__license__ = "See LICENSE.txt file"
-__version__ = "0.1.0"
-__maintainer__ = "Alexis Guyot"
-__email__ = "alexis.guyot@icm-institute.org"
-__status__ = "Development"
 
 import clinica.pipelines.engine as cpe
 
 
 class T1FreeSurferTemplate(cpe.Pipeline):
     """FreeSurfer Longitudinal template class
 
-    Creates a pipelines that runs the FreeSurfer longitudinal template
-    initialisation module for each subjects in a .tsv-defined list of
-    subjects/sessions. This requires a prior-run of
-    t1-freesurfer on the .tsv.
-
-    Todos: N/A
-
     Returns:
         A clinica pipeline object containing the T1FreeSurferTemplate pipeline.
-
-    Raises:
-
     """
 
+    @staticmethod
+    def get_processed_images(caps_directory, subjects, sessions):
+        import os
+        from clinica.utils.inputs import clinica_file_reader
+        from clinica.utils.input_files import T1_FS_T_DESTRIEUX
+        from clinica.utils.longitudinal import get_long_id
+        from clinica.utils.participant import get_unique_subjects
+        import re
+
+        [list_participant_id, list_list_session_ids] = get_unique_subjects(subjects, sessions)
+        list_long_id = [get_long_id(list_session_ids) for list_session_ids in list_list_session_ids]
+
+        image_ids = []
+        if os.path.isdir(caps_directory):
+            t1_freesurfer_files = clinica_file_reader(
+                list_participant_id, list_long_id,
+                caps_directory, T1_FS_T_DESTRIEUX, False
+            )
+            image_ids = [re.search(r'(sub-[a-zA-Z0-9]+)_(long-[a-zA-Z0-9]+)', file).group()
+                         for file in t1_freesurfer_files]
+        return image_ids
+
     def check_pipeline_parameters(self):
```

Comment:
Why we keep this (unnecessary) function in most of the pipelines? Is it supposed to be completed later?

Reply:
This function is automatically called when the pipeline is built and, as a result, run.

It was designed for the checking of pipeline parameters (see e.g. [StatisticsSurface](https://github.com/aramis-lab/clinica/blob/dev/clinica/pipelines/statistics_surface/statistics_surface_pipeline.py)). This can be also used for setting default values of a pipeline when instanciating a new pipeline outside the CLI (e.g. with the continuous integration). See e.g. [T1VolumeTissueSegmentation](https://github.com/aramis-lab/clinica/blob/dev/clinica/pipelines/t1_volume_tissue_segmentation/t1_volume_tissue_segmentation_pipeline.py). 

It was implemented in several pipeline but I don't remember if this is done for **all** pipelines.

Of course, if a pipeline has no parameter (e.g. DwiDti), this is useless :)